File: src/DataTypes/MyList.java
package DataTypes;

import java.util.LinkedList;
import Exception.EmptyMyListException;

/**
 * Created by vladc on 22.10.2016.
 */
public class MyList<T> implements MyIList<T> {
    private LinkedList<T> queue;

    public MyList() {
        this.queue = new LinkedList<T> ();
    }

    public void enqueue(T e) {
        this.queue.addFirst(e);
    }

    public T dequeue() throws EmptyMyListException {
        if (queue.isEmpty())
            throw new EmptyMyListException("Exception. MyList is empty.");
        return queue.removeLast();
    }

    public boolean isEmpty() {
        return queue.isEmpty();
    }

    @Override
    public String toString() {
        return queue.toString();
    }
}

File: Toy-Language-Interpreter_C#/Toy-Language-Interpreter/DataTypes/MyList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.MyException;

namespace Toy_Language_Interpreter.DataTypes
{
    class MyList<T> : MyIList<T>
    {
        private List<T> list;

        public MyList()
        {
            this.list = new List<T>();
        }

        public void Enqueue(T e)
        {
            this.list.Add(e);
        }

        public T Dequeue()
        {
            if (list.Count == 0)
                throw new EmptyMyListException("Exception. MyList is empty.");
            T ret = list.ElementAt(0);
            list.RemoveAt(0);
            return ret;
        }

        public bool IsEmpty()
        {
            return list.Count == 0;
        }

        public override string ToString()
        {
            string str = "";

            foreach (T t in list)
            {
                str += t.ToString();
                str += "\n";
            }
            return str;
        }
    }
}
